tg_bot/modules/locks.py

```diff
@@ -80,6 +83,59 @@ def locktypes(bot: Bot, update: Update):
     update.effective_message.reply_text("\n - ".join(["Locks: "] + list(LOCK_TYPES) + list(RESTRICTION_TYPES)))
 
 
+@user_admin
+def add_whitelist(bot: Bot, update: Update):
+    chat = update.effective_chat  # type: Optional[Chat]
+    message = update.effective_message  # type: Optional[Message]
+    entities = message.parse_entities(MessageEntity.URL)
+    added = []
+    for url in entities.values():
+        if sql.add_whitelist(chat.id, url):
+            added.append(url)
+    if added:
+        message.reply_text("Added {} to whitelist.".format(', '.join(w for w in added)))
+    else:
+        message.reply_text("No URLs were added to the whitelist")
+
+
+@user_admin
+def remove_whitelist(bot: Bot, update: Update):
+    chat = update.effective_chat  # type: Optional[Chat]
+    message = update.effective_message  # type: Optional[Message]
+    entities = message.parse_entities(MessageEntity.URL)
+    removed = []
+    for url in entities.values():
+        if sql.remove_whitelist(chat.id, url):
+            removed.append(url)
+    if removed:
+        message.reply_text("Removed `{}` from whitelist.".format('`, `'.join(escape_markdown(w) for w in removed)),
+            parse_mode=ParseMode.MARKDOWN)
+    else:
+        message.reply_text("Could not remove URL from whitelist or URL not found.")
+
+def list_white(bot: Bot, update: Update):
+    chat = update.effective_chat  # type: Optional[Chat]
+    message = update.effective_message  # type: Optional[Message]
+    all_whitelisted = sql.get_whitelist(chat.id)
+
+    if not all_whitelisted:
+        message.reply_text("No URLs are whitelisted here!")
+        return
+
+    BASIC_WHITE_STRING = "Whitelisted URLs:\n"
+    listwhite = BASIC_WHITE_STRING
+    for url in sorted(all_whitelisted.keys()):
+        entry = "{}, ".format(url)
+        if len(entry) + len(listwhite) > MAX_MESSAGE_LENGTH:
```

Comment:
There already is a split_message function available for longer messages, see the helpers package